src/myn/patterns/decorator/GraphicRedactorDecorator/AbstractShape.java

```diff
@@ -2,26 +2,30 @@
 
 /**
  * @author Zaichenko Ivan
- * Abstract class AbstractShapeRedactor for work of shape
+ * Abstract class AbstractShape for work of shape
  */
-public abstract class AbstractShapeRedactor {
+public abstract class AbstractShape {

```

Comment:
Do you realize that every decorator has a copy of `x`, `y`, and `size` that they don't use?
And `Shape` has `shapeClone` that it does not use.
That's *24 bytes* per decorator (if memory serves me right) and *8 bytes* per Shape.